platform/mbed_retarget.h

```diff
@@ -619,6 +624,31 @@ std::FILE *fdopen(mbed::FileHandle *fh, const char *mode);
  */
 int bind_to_fd(mbed::FileHandle *fh);
 
+#else
+/** Targets may implement this to override how to write to the console.
+ *
+ * If the target hasn't provided minimal_console_putc, this is called
+ * to give the target a chance to specify a serial access to the console.
```

Comment:
"a serial access"? It's to specify an alternative minimal console - so may well not be serial.